10x/wdls/count.wdl

```diff
@@ -0,0 +1,1072 @@
+# )
+task extract_reads_split {
+  File chunks
+  String barcode_whitelist
+  
+  String d = "$"
+
+  command {
+
+    chunks_string=${d}(<${chunks})
+
+    /opt/tenx/run_in_10x_env.bash martian stage run EXTRACT_READS split \
+      --chunks "${d}chunks_string" \
+      --barcode_whitelist ${barcode_whitelist}
+  }
+ 
+  runtime {
+    docker: "10x_hca_wdl:latest"
+  }
+
+  output {
+    Array[File] split_files = glob("martian_split_*")
+  }
+}
+
+task extract_reads_main {
+  File chunks
+  File chemistry_def
+  String barcode_whitelist
+  Int reads_per_file
+  Float subsample_rate
+  Array[Map[String, String]] primers
+
+  File split_file
+  Array[File] fastq_chunks
+ 
+  String d = '$' 
+  String l = '{'
+  String r = '}'
+ 
+  command {
+
+    # Put the fastq file into the $PWD/files directory where martian is expecting
+    # them
+    mkdir files
+    for fastq_chunk in ${sep=" " fastq_chunks}; do
+      ln "${d}fastq_chunk" files/"${d}(basename ${d}fastq_chunk)"
+    done
+    
+    # Read the chunks and chemistry json files to strings so we can pass them as
+    # a Martian map
+    chunks_string=${d}(<${chunks})
+    chemistry_def_string=${d}(<${chemistry_def})
+
+    /opt/tenx/run_in_10x_env.bash martian stage run EXTRACT_READS main \
+      --chunks "${d}chunks_string" \
+      --chemistry_def "${d}chemistry_def_string" \
+      --barcode_whitelist ${barcode_whitelist} \
+      --reads_per_file '${reads_per_file}' \
+      --subsample_rate '${subsample_rate}' \
+      --skip_metrics no \
+      --primers '[${sep="," primers}]' \
+      --split_file '${split_file}'
+    
+    # The Martian cli adapter dumps these outputs as json file, so convert them
+    # to a format Cromwell can parse
+    jq -r '.[]' read_groups > read_groups.lines
+    jq -r '.[]' gem_groups > gem_groups.lines
+    jq -r '.[]' bam_comments > bam_comments.lines
+
+  }
+
+  runtime {
+    docker: "10x_hca_wdl:latest"
+  }
+
+  output {
+    Array[File] trimmed_seqs = glob("trimmed_seqs.bam/*")
+    Array[File] reads = glob("reads.fastq/*")
+    File barcode_counts = "barcode_counts.json"
+    File chunked_reporter = "chunked_reporter.pickle"
+    Array[String] read_groups = read_lines("read_groups.lines")
+    Array[Int] gem_groups = read_lines("gem_groups.lines")
+    Array[String] bam_comments = read_lines("bam_comments.lines")
+    File align = "align"
+  }
+}
+
+task extract_reads_join {
+  String align 
+  String barcode_whitelist
+  File chemistry_def
+  Array[Array[File]] reads_output
+  Array[Array[File]] trimmed_seqs_output
+  Array[Array[String]] bam_comments_output
+  Array[Array[Int]] gem_groups_output
+  Array[Array[String]] read_groups_output
+  Array[File] chunked_reporter_output
+  Array[File] barcode_counts_output
+    
+  Array[File] split_files
+  
+  String d = "$"
+  String l = "{"
+  String r = "}"
+
+  command {
+    set -x  
+    
+    # There's a lot here, but it's mostly the same thing repeated. All the input reads and trimmed_seq
```

Comment:
Given that it's mostly the same thing repeated, is there any way to improve this? could you define a function or subtask that gets called here to do this to reduce the amount of repeated code?